actionview/lib/action_view/helpers/form_options_helper.rb

```diff
@@ -776,18 +805,20 @@ def option_text_and_value(option)
           end
         end
 
-        def option_value_selected?(value, selected)
-          Array(selected).include? value
-        end
-
-        def extract_selected_and_disabled(selected)
+        def extract_select_options(selected)
           if selected.is_a?(Proc)
-            [selected, nil]
+            [selected, nil, nil, nil, nil]
```

Comment:
I'm not sure how important the # of nils are, depending on how the consumers use the array, but it's possible you can skip out on the extra ones (or even remove the existing one) and save some memory due to how ruby arrays work
```ruby
[12] pry(main)> test = []
=> []
[13] pry(main)> test.length
=> 0
[14] pry(main)> test[5]
=> nil
[15] pry(main)> test = [nil, nil, nil, nil, nil]
=> [nil, nil, nil, nil, nil]
[16] pry(main)> test.length
=> 5
[17] pry(main)> test[5]
=> nil
```